x-pack/plugins/lens/public/xy_visualization/expression.tsx

```diff
@@ -206,6 +214,37 @@ export const getXyChartRenderer = (dependencies: {
   },
 });
 
+function mergeThemeWithValueLabelsStyling(
+  theme: RecursivePartial<Theme>,
+  valuesLabelMode: string = 'hide',
+  isHorizontal: boolean
+) {
+  const VALUE_LABELS_FONTSIZE = 15;
+
+  if (valuesLabelMode === 'hide') {
+    return theme;
+  }
+  return {
+    ...theme,
+    ...{
+      barSeriesStyle: {
+        ...theme.barSeriesStyle,
+        displayValue: {
+          fontSize: { min: 10, max: VALUE_LABELS_FONTSIZE },
+          fill: { textInverted: true, textBorder: true },
+          alignment: isHorizontal
+            ? {
+                vertical: VerticalAlignment.Middle,
+              }
+            : { horizontal: HorizontalAlignment.Center },
+          offsetX: isHorizontal ? 10 : 0,
+          offsetY: isHorizontal ? 0 : -5,
```

Comment:
This might be a personal feeling, but IMHO the labels are too close to the upper edge of the bar - can we go to -10 or something here?